tests/test_api_requests.py

```diff
@@ -213,6 +215,47 @@ async def test_job_id(self, job_id):
             assert "zyte-api" in resp.flags
             assert resp.body == f"<html>{job_id}</html>".encode("utf8")
 
+    @pytest.mark.parametrize(
+        "method",
+        [
+            "DELETE",
+            "GET",
+            "OPTIONS",
+            "PATCH",
+            "POST",
+            "PUT",
+            "TRACE",
+            # Not supported by Zyte Data API at the moment, but processed
+            # nonetheless for forward-compatibility, in case Zyte Data API
+            # starts supporting any of them in the future:
```

Comment:
Thanks for noting these down! 

Currently, whenever we send the `httpRequestMethod` values below, Zyte API returns something like:
```
{"type":"/request/invalid","title":"Bad Request","status":400,"detail":"Cannot construct instance of `com.zyte.gen.model.ExtractRequest$HttpRequestMethodEnum`, problem: Unexpected value 'CONNECT'\n at [Source: (org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$UnCloseableInputStream); line: 1, column: 87] (through reference chain: com.zyte.gen.model.ExtractRequest[\"httpRequestMethod\"])"}
```

We'll need to provide a better error message (or at least a warning) when such `httpRequestMethod` values are used. 

On second thought, I think a much better approach would be to block the request from the client-side completely. This also prevents having the user commit an unnecessary and erroneous request.

Reply:
> I think a much better approach would be to block the request from the client-side completely.

I agree as long as the server team confirms that they do not ever plan to support those, i.e. that not supporting them is very explicit, and that they are not willing to add support for them in the future. I imagine that is the case for `CONNECT`, but I am not so sure about `HEAD`.